### PR TITLE
FIX: dereference nullptr

### PR DESCRIPTION
**Pull request by RavenX8**

### CODE REVIEW

**Comment by L3nn0x** [comment]
There is a problem with your fix : first you still have to close the socket and you also have to decrement the clientConnected variable before exiting the function.

File: Common/src/SocketServer.cpp [hunk 1]
```diff
@@ -394,13 +394,10 @@ void CServerSocket::HandleClients( fd_set* fds )
...
+	ConnectedClients++;
 	CClientSocket* thisclient = this->CreateClientSocket( );
-	if (thisclient==NULL) 
-    { 
-		closesocket( thisclient->sock );
-		if (thisclient!=0) delete thisclient;
-		thisclient=0;
+	if ( thisclient == NULL )
+	{
 		return;
```
Comment (L3nn0x) [inline Common/src/SocketServer.cpp:403]: You have to add before exiting :
closesocket(sock);
--ConnectedClients;

Comment (RavenX8) [inline Common/src/SocketServer.cpp:403]: You can't close the socket as your unable to use -> on thisclient since its null.

**Comment by RavenX8** [comment]
Fixed the connected client issue.

You can't close the socket since it was never created. If CClientSocket\* thisclient = this->CreateClientSocket( ); returns null, the socket was never created because there was an issue allocating the memory to do so.

**Comment by RavenX8** [comment]
Actually I see what your saying. It was a bug in the first place. It should of been calling closesocket on sock not thisclient->sock

**Comment by L3nn0x** [comment]
Yeah it was a bug in the first place. Nice finding that one.

**Comment by peterson2017** [comment]
RavenX8 Poderia me passer seu contato por favor , gostaria de converssar um assunto.

**Pull request merged**